src/CMakeLists.txt

```diff
@@ -8,7 +10,8 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 find_package(RT REQUIRED)
 find_package(DL REQUIRED)
 find_package(M REQUIRED)
-find_package(GLIB REQUIRED)
+
+pkg_check_modules(GLIB REQUIRED glib-2.0>=2.32)
```

Comment:
Hmm. `CMAKE_EXTRA_LIBRARIES` and `CMAKE_EXTRA_INCLUDES` appear to be non-standard; they only work for the custom search modules that are still included (e.g. FindDL.cmake).

For pkg-config specifically, the only way I see documented in cmake for extending the search path is [CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html#variable:CMAKE_PREFIX_PATH). I added a `--search` option to the setup sript to set that variable.

As a bonus, I *think* this could also be used in place of `CMAKE_EXTRA_LIBRARIES` and `CMAKE_EXTRA_INCLUDES`. I thought about ripping out those and the corresponding setup script flags, but I don't have a convenient way to test it. Maybe it's still useful to be able to override those more specifically anyway.

Bouncing back for another review round

Reply:
Could this `CMAKE_PREFIX_PATH` be added to the custom search modules so that `--search` can be used for everything? For example:

```cmake
find_path (DL_INCLUDES dlfcn.h
  PATHS /usr/local/include /usr/include ${CMAKE_PREFIX_PATH}/include ${CMAKE_EXTRA_INCLUDES}
  )

find_library (DL_LIBRARIES dl
  PATHS /usr/local/lib /usr/lib /lib ${CMAKE_PREFIX_PATH}/lib ${CMAKE_EXTRA_LIBRARIES}
  )
```

And we could keep `--library` and/or `--include` around for legacy reasons?

I think it is a little confusing having three options that work for different libraries (glib requires `--search` and everything else requires `--library` and/or `--include`), but I also don't have a simple way to test it, so I think it's fine since they're probably not used often.

Reply:
> Could this CMAKE_PREFIX_PATH be added to the custom search modules so that --search can be used for everything?

Yes, it already works for those; no need to add it explicitly in the `find_path` and `find_library` invocations. e.g. from [find_path](https://cmake.org/cmake/help/latest/command/find_path.html#command:find_path)

> <prefix>/include for each <prefix> in [CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html#variable:CMAKE_PREFIX_PATH)

> I think it is a little confusing having three options that work for different libraries (glib requires --search and everything else requires --library and/or --include), but I also don't have a simple way to test it, so I think it's fine since they're probably not used often, and over time we'll have fewer and fewer C libraries.

Yeah, I was thinking about ripping those out, but it's difficult to test. The only things we're using them for are sub-libraries of libc, and other things break if libc isn't already in a standard location or LIBRARY_PATH and C_INCLUDE_PATH, which seems to already be part of the default searches.

Maybe I'll go ahead and mark them deprecated and drop from the docs though, and we can removing them completely. (Maybe in the next major release)